### PR TITLE
blacklisted props

### PR DESCRIPTION
**Pull request by Dacello**

### CODE REVIEW

File: lib/execjs/rails/engine.rb [hunk 1]
```diff
@@ -12,7 +12,16 @@ class Engine < ::Rails::Engine
       config.execjs_rails.has_view_function_name = 'execjs_rails_has_view'
       config.execjs_rails.handler_function_name = 'execjs_rails_handler'
 
+      config.execjs_rails.blacklisted_assigns = [
+        'devise_mapping', 'devise_parameter_sanitizer'
+      ]
+
       config.execjs_rails.view_to_function_options = lambda do |view, options|
+        # reject keys in the blacklist
+        view.assigns.reject! do |key|
```
Comment (jwietelmann) [inline lib/execjs/rails/engine.rb]: `reject!` mutates the existing hash I think. We don't want that. We want a *copy* of `view.assigns` with the removed keys.

File: lib/execjs/rails/engine.rb [hunk 2]
```diff
@@ -12,7 +12,16 @@ class Engine < ::Rails::Engine
       config.execjs_rails.has_view_function_name = 'execjs_rails_has_view'
       config.execjs_rails.handler_function_name = 'execjs_rails_handler'
 
+      config.execjs_rails.blacklisted_assigns = [
+        'devise_mapping', 'devise_parameter_sanitizer'
```
Comment (jwietelmann) [inline lib/execjs/rails/engine.rb]: We should probably just make this empty and let stuff like `revelry_core` add to it.

File: lib/execjs/rails/engine.rb (hunk 1 above)
Comment (Dacello) [inline lib/execjs/rails/engine.rb]: Ok, didnt know we didnt want to mutate the existing hash. Out of curiosity, why dont we want to mutate the hash directly?

File: lib/execjs/rails/engine.rb (hunk 2 above)
Comment (Dacello) [inline lib/execjs/rails/engine.rb]: that makes sense.

File: lib/execjs/rails/engine.rb (hunk 1 above)
Comment (jwietelmann) [inline lib/execjs/rails/engine.rb]: I mean, it *probably* doesn't really matter, but there's no point in creating side-effects when we don't need them. The whole point is to take part of the view context and serialize it as JSON, not manipulate the actual view context.

Comment (Dacello) [inline lib/execjs/rails/engine.rb]: ok cool. was jw

**Pull request merged**